c/example04.c

```diff
@@ -0,0 +1,148 @@
+/* Example 04 :
+This example shows how to create and delete a tag for a specified resource.
+*/
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <curl/curl.h>
+
+#define AK_SIZE 20
+#define SK_SIZE 40
+
+#define auto_tag  __attribute__ ((__cleanup__(delete_tag_cleanup)))
+
+struct Tag {
+    char *resource_id;
+    char *key;
+    char *value;
+    CURL *curl;
+    struct curl_slist *headers;
```

Comment:
I think, you can init the header list in the main too, and remove this from the struct.
(so you call `curl_easy_setopt(tag->curl, CURLOPT_HTTPHEADER, tag->headers);` in main)